relentless/ensemble.py

```diff
@@ -28,94 +73,96 @@ def __init__(self, types, T, P=None, V=None, mu={}, N={}, kB=1.0, conjugates=Non
             raise ValueError('Either P or V must be set.')
 
         # mu-N, must be set by type
-        self._mu = core.TypeDict(self.types)
-        self._N = core.TypeDict(self.types)
+        self._mu = core.FixedKeyDict(keys=self.types)
+        self._N = core.FixedKeyDict(keys=self.types)
         for t in self.types:
             if (t not in mu or mu[t] is None) and (t not in N or N[t] is None):
                 raise ValueError('Either mu or N must be set for type {}.'.format(t))
+            if t in N and N[t] is not None and not isinstance(N[t], int):
+                raise TypeError('All values of N must be integers.')
             if t in N:
                 self._N[t] = N[t]
             if t in mu:
                 self._mu[t] = mu[t]
```

Comment:
This logic is a little hard to read. You could probably just set these using the `update` methods of the dicts, and then just check the things you need. The types are checked below, so I think all you need to add is that N is all ints.